BAU: Bump express to v5

alhcomer opened the pull request:
## What

- Bumps the npm-express-dependencies group with 3 updates in the / directory: [express](https://github.com/expressjs/express), [@types/express](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express) and [express-validator](https://github.com/express-validator/express-validator).

- See change descriptions in commit messages

## How to review

1. Code Review commit by commit
2. Run through journey, go to create account, submit empty email, see validation error
3. Deploy to an environment and run acceptance tests against branch

## Checklist

<!-- Performance analysis notice
Make sure that a performance analyst colleague in your team has been informed of any changes to the user interfaces or user journeys.

Delete this item if the PR does not change any UI or user journeys.
-->

- [ ] Performance analyst has been notified of the change.

<!-- UCD review
Changes to the user journeys, the UI or content should be reviewed by Content Design and Interaction Design before being merged.

This is to ensure:
- They have an opportunity to confirm the implementation meets expectations.
  - For example, how error screens appear and whether invalid entries can be amended.
- They can make any necessary changes to Figma designs.

It is also important that new features have a full end-to-end journey review before going live. Ensure this is done before enabling a feature flag that changes the frontend.

Contact UCD colleagues in the Authentication team to identify the best way to approach the review.

If including screenshots in the PR description, include those representing error states. Here are some examples:
- a PR that uses tables to display the screenshots https://github.com/govuk-one-login/authentication-frontend/pull/1187
- a PR that uses a dropdown to display the screenshots https://github.com/alphagov/di-infrastructure/pull/578

You can find example PRs from this repository that include screenshots through this search: https://github.com/govuk-one-login/authentication-frontend/pulls?q=is%3Apr+screenshots+is%3Aclosed+is%3Amerged

Delete this item if the PR does not change the UI.
-->

- [ ] A UCD review has been performed.

<!-- Acceptance tests have been updated
This is to avoid failures occurring after a merge. The types of changes that may impact acceptance tests might be:

- changes to user journeys
- changes to the text of page titles
- changes to the text of interactive elements (such as links).

The Test Engineers on the Authentication Team will be happy to discuss any changes if you're unsure.
-->

- [ ] Any necessary changes to the [acceptance tests](https://github.com/govuk-one-login/authentication-acceptance-tests) have been made.

<!-- Associated documentation has been updated
This might include updates to the README.md, Confluence pages etc.
-->

- [ ] Documentation has been updated to reflect these changes.

## Related PRs

<!-- Links to PRs in other repositories that are relevant to this PR.

This could be:
  - PRs that depend on this one
  - PRs this one depends on
  - If this work is being duplicated in other repos, other PRs
  - PRs which just provide context to this one. -->

<!-- Delete this section if not needed! -->

dependabot[bot] committed:
BAU: Bump the npm-express-dependencies group across 1 directory with 3 updates

Bumps the npm-express-dependencies group with 3 updates in the / directory: [express](https://github.com/expressjs/express), [@types/express](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express) and [express-validator](https://github.com/express-validator/express-validator).

Updates `express` from 4.21.2 to 5.1.0
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.21.2...v5.1.0)

Updates `@types/express` from 4.17.21 to 5.0.3
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/express)

Updates `express-validator` from 6.13.0 to 7.2.1
- [Release notes](https://github.com/express-validator/express-validator/releases)
- [Commits](https://github.com/express-validator/express-validator/compare/v6.13.0...v7.2.1)

Updates `@types/express` from 4.17.21 to 5.0.3
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/express)

---
updated-dependencies:
- dependency-name: express
  dependency-version: 5.1.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: npm-express-dependencies
- dependency-name: "@types/express"
  dependency-version: 5.0.3
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: npm-express-dependencies
- dependency-name: express-validator
  dependency-version: 7.2.1
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: npm-express-dependencies
- dependency-name: "@types/express"
  dependency-version: 5.0.3
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: npm-express-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

alhcomer committed:
BAU: validation error formatter

- Throw for unsupported error types (alternative, alternative_grouped, unknown_fields)
  as these are unused in the project

alhcomer committed:
BAU: remove express-async-errors package

- Async error handling is included out of the box in express 5

alhcomer committed:
BAU: update wildcard route for sanitizeRequestMiddleware

- Replace '*' with '/*splat' in sanitize middleware route to fix
  path-to-regexp "Missing parameter name" error.
- Express v5 uses stricter route validation that requires named
  parameters for wildcard patterns.
- See guidance for path syntax (and splat) here:
  https://expressjs.com/en/guide/migrating-5.html#path-syntax

alhcomer committed:
BAU: fix overloadProtection app.ts tests

- Express no longer exposes the stack property as a public API
- Use fakes instead to assert that overloadProtection is used
  conditionally depending on environment

alhcomer committed:
BAU: mock getVitalSignsIntervalSeconds function

- This is so that the function returns the same value in app.test.ts
  regardless of what value is set for VITAL_SIGNS_INTERVAL_SECONDS in
  the .env file

alhcomer committed:
BAU: Don't reference req.body.isResendCodeRequest in GET routes

Express v5 changed how req.body is handled for GET requests:
- Express v4: req.body was {} for GET requests, accessing undefined properties returned undefined
- Express v5: req.body is undefined for GET requests, accessing properties throws TypeError

The uplift journey is a GET request but the sendMfaGeneric function was
accessing req.body.isResendCodeRequest without null checking, causing:
"Cannot read properties of undefined (reading 'isResendCodeRequest')"

Changes:
- Use optional chaining (req.body?.isResendCodeRequest ?? false) for safe access

This maintains the same behavior (isResendCodeRequest = false for GET requests)
while being compatible with Express v5's stricter undefined handling.

alhcomer committed:
BAU: fix contact us bugs exposed by express v5 bump

- Guards against undefined req.body on get requests
- Ensures SmartAgent config singleton uses mocked URL in
  integration tests

alhcomer commented on src/app.ts:
```diff
@@ -297,7 +296,7 @@ async function createApp(): Promise<express.Application> {
   app.use(environmentBannerMiddleware);
   app.use(getSessionIdMiddleware);
-  app.post("*", sanitizeRequestMiddleware);
+  app.post("/*splat", sanitizeRequestMiddleware);
```
splat is the wildcard param name express mention to use in the [v5 migration guide](https://expressjs.com/en/guide/migrating-5.html#:~:text=node%20index.js-,Changed,behavior%20of%20parameters%20%3A%2C%20use%20/*splat%20instead%20of%20/*,-//%20v4%0Aapp.)

alhcomer committed:
BAU: delete yarn.lock

BeckaL commented on src/middleware/form-validation-middleware.ts:
```diff
@@ -1,18 +1,31 @@
+      return `Unknown fields found, please remove them: ${fields}`;
+    }
+    default:
+      throw new Error(`Not a known express-validator error: ${error}`);
```
This interpolated error will get serialised as [object Object]

BeckaL commented on src/middleware/form-validation-middleware.ts:
```diff
@@ -1,18 +1,31 @@
+      };
+    case "alternative":
+    case "alternative_grouped":
+      return error.msg;
```
We have an issue here that this type is now different to the type that renderBadRequest expects.

As far as I understand it, these errors can't currently be thrown in our codebase since they're for when oneOf validation is used, and we don't use it currently in this codebase. An alternative here might be to throw an error for these cases, and force it to be handled explicitly in the associated change PR if we ever start using these validations. I think I'd prefer this rather than shaping the return type here to something expected, since likely if we got to the point of implementing this, we'd want to think about whether the types here work or not / it'd be easier to work from a concrete example.

BeckaL commented on src/middleware/form-validation-middleware.ts:
```diff
@@ -1,18 +1,31 @@
+      return error.msg;
+    case "unknown_fields": {
+      const fields = error.fields.map((field) => field.path).join(", ");
+      return `Unknown fields found, please remove them: ${fields}`;
```
Same issue here. From some reading, this error case is for if we were using [checkExact](https://express-validator.github.io/docs/api/check-exact/), but we don't currently. Even if we were, it would potentially make more sense to throw an error.

alhcomer committed:
BAU: Update .secrets.baseline

- Ran `detect-secrets scan --baseline .secrets.baseline`
- Followed by `detect-secrets audit .secrets.baseline`, wherein
  I've marked flagged strings as false positives. They were
  either test values, strings with words like "secret", or safe
  AWS ARNs
